### PR TITLE
Support `retentionTimeout` and `waitForPodSec`

### PR DESCRIPTION
**Pull request by AhmadMasry**
### What does this PR do?

Add the ability to modify `retentionTimeout` and `waitForPodSec` default value in JCasC:
- retentionTimeout: For tasks that use very large images, this timeout can be increased to avoid early termination of the task while the Kubernetes pod is still deploying.
- waitForPodSec: This will change how long Jenkins will wait (seconds) for pod to be in running state.

```[tasklist]
### Submitter checklist
- [x] I bumped the "version" key in `./charts/jenkins/Chart.yml`.
- [x] I added a new changelog entry to `./charts/jenkins/CHANGELOG.md`.
- [x] I followed the [technical requirements](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#technical-requirements).
```

### CODE REVIEW

**Comment by AhmadMasry** [comment]
Is there anything I can do to fix the tests or to go forward?
Thank you

File: charts/jenkins/README.md [hunk 1]
```diff
@@ -297,6 +297,22 @@ agent:
 ```
 This will change the configuration of the kubernetes "cloud" (as called by jenkins) that is created automatically as part of this helm chart.
 
+### Change container cleanup cimeout API
```
Comment (timja) [inline charts/jenkins/README.md:300]:
```suggestion
### Change container cleanup timeout API
```

**Comment by timja** [comment]
> Is there anything I can do to fix the tests or to go forward? Thank you

You need to update the checksums, the new ones are in the logs, search the string that is failing in the test and replace it with the new one

File: charts/jenkins/README.md (hunk 1 above)
Comment (AhmadMasry) [reply charts/jenkins/README.md]: Add, thank you

**Comment by AhmadMasry** [comment]
@timja Thank you for you review.
I fixed your suggestion and fixed all tests.

**Review by timja — APPROVED** [review]

**Pull request merged**